modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/modal/Modal.js

```diff
@@ -78,6 +78,21 @@ const Modal = ({
 	});
 
 	const onButtonClick = ({formId, onClick, type}) => {
+		const submitForm = (form) => {
+			if (form.requestSubmit) {
+				form.requestSubmit();
+			}
+			else {
+				const canceled = form.dispatchEvent(
+					new Event('submit', {cancelable: true})
+				);
+
+				if (canceled) {
+					form.submit();
+				}
+			}
+		};
+
```

Comment:
TIL `requestSubmit` exists. Nice!

From the [docs](https://html.spec.whatwg.org/multipage/forms.html#dom-form-requestsubmit), it is not clear to me what happens when there is no argument. 
> Otherwise, set submitter to this [form](https://html.spec.whatwg.org/multipage/forms.html#the-form-element) element.

Does it still do the validation and throw the `submit` event? If not, we may query `input[type=submit]`, and set that as argument.

If it works as expected, we don't need the hidden `<input>`.

Reply:
Hey @markocikos without the argument the validation works as expected :) 

The hidden input is required to allow the form submission when clicking enter in any input

Reply:
So, `form.submit()` does not require button, but `form.requestSubmit()` requires it? That's weird :thinking: